torrent/tracker.go

```diff
@@ -206,19 +208,24 @@ func (self *Tracker) serveFile(w http.ResponseWriter, r *http.Request, file *Fil
 		// TODO: This could run infinitely in a case where the file is requested and deleted or
 		// replaced, so we keep checking a structure that never will get filled in since it's no
 		// longer active.
+		file.Lock.Lock()
```

Comment:
Hah, ownership only gives me buttons to press. I'd be a bad maintainer if I didn't listen to feedback.

I'm happy to go with the simple solution and we'll find out in prod if it doesn't work out, and in that case, we can revisit.